### PR TITLE
[MDS-5976] AMS Payment contact info

### PR DESCRIPTION
**Pull request by henryoforeh-dev**
## Objective 

[MDS-5976](https://bcmines.atlassian.net/browse/MDS-5976)

_Why are you making this change? Provide a short explanation and/or screenshots_

- Created component to store payment contact information.
- Added migration script to create new column in project summary table and type code for mine payment party. 

image

### CODE REVIEW

File: services/common/src/components/projectSummary/PaymentContact.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  const formValues = useSelector(getFormValues(FORM.ADD_EDIT_PROJECT_SUMMARY));
+  const { payment_contact = {} } = formValues;
+  const provinceOptions = useSelector(getDropdownProvinceOptions);
+
+  useEffect(() => {
+    dispatch(change(FORM.ADD_EDIT_PROJECT_SUMMARY, "payment_contact.party_type_code", "PER"));
+  }, []);
+
+  return (
+    <div className="ant-form-vertical" style={{ paddingTop: 12 }}>
```
Comment (taraepp) [inline services/common/src/components/projectSummary/PaymentContact.tsx]: Shouldn't need the "ant-form-vertical" classname, the parent already has it.

File: services/common/src/components/projectSummary/PaymentContact.tsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        </Col>
+      </Row>
+
+      <Row gutter={16}>
+        <Col md={12} sm={24}>
+          <Field name="payment_contact.address[0].city" label="City" component={RenderField} />
+        </Col>
+        <Col md={12} sm={24}>
+          <Field
+            name="payment_contact.address[0].post_code"
```
Comment (taraepp) [inline services/common/src/components/projectSummary/PaymentContact.tsx:126]: I'm wondering about the validation on these. This one looks like it wants the `postalCodeWithCountry` validation, and is the country not even required? I think it autosaves with Canada if sent through as null.

File: services/common/src/components/projectSummary/PaymentContact.tsx (hunk 1 above)
Comment (henryoforeh-dev) [reply services/common/src/components/projectSummary/PaymentContact.tsx]: Updated.

File: services/common/src/components/projectSummary/PaymentContact.tsx (hunk 2 above)
Comment (henryoforeh-dev) [reply services/common/src/components/projectSummary/PaymentContact.tsx:126]: Added validation. On the figma design, all the address fields are optional. I think it is also in line with what Ana asked for during our meeting with her. The main information they want are the ones that are mandatory. 

**Figma design screenshot.**
https://www.figma.com/design/O7i6jpbKbju49HfL1OpozT/CORE-%26-Minespace-2021-2023?node-id=30995-90032&t=0utC7RRmtI5R9Cfj-0

image

**Review by matbusby-fw — DISMISSED** [review]

**Review by matbusby-fw — APPROVED** [review]

**Review by taraepp — APPROVED** [review]

**Pull request merged**